## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 502299`
Commit: [`3e820e7dc300f13d2e7e3c8c20b317f8b2ae1530`](https://github.com/NixOS/nixpkgs/commit/3e820e7dc300f13d2e7e3c8c20b317f8b2ae1530) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/3e820e7dc300f13d2e7e3c8c20b317f8b2ae1530..pull/502299/head))
Merge: [`e3c01d01322142e71b095a01f6da1e0fc7bdf90d`](https://github.com/NixOS/nixpkgs/commit/e3c01d01322142e71b095a01f6da1e0fc7bdf90d)

Logs: https://github.com/JamieMagee/nixpkgs-review-gha/actions/runs/23414602005


---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>tests.home-assistant-component-tests.hypontech</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.hyponcloud</li>
    <li>python313Packages.hyponcloud.dist</li>
    <li>python314Packages.hyponcloud</li>
    <li>python314Packages.hyponcloud.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>tests.home-assistant-component-tests.hypontech</summary>
<pre>2026-03-22 23:05:33.493 INFO     MainThread homeassistant.helpers.entity_registry:entity_registry.py:1260 Registered new sensor.hypontech entity: sensor.rooftop_power
2026-03-22 23:05:33.494 INFO     MainThread homeassistant.helpers.entity_registry:entity_registry.py:1260 Registered new sensor.hypontech entity: sensor.rooftop_lifetime_energy
2026-03-22 23:05:33.495 INFO     MainThread homeassistant.helpers.entity_registry:entity_registry.py:1260 Registered new sensor.hypontech entity: sensor.rooftop_today_energy
=============================== warnings summary ===============================
homeassistant/runner.py:176
  /build/source/homeassistant/runner.py:176: DeprecationWarning: &#x27;asyncio.DefaultEventLoopPolicy&#x27; is deprecated and slated for removal in Python 3.16
    class HassEventLoopPolicy(asyncio.DefaultEventLoopPolicy):  # type: ignore[name-defined,misc]

tests/conftest.py:155
  /build/source/tests/conftest.py:155: DeprecationWarning: &#x27;asyncio.set_event_loop_policy&#x27; is deprecated and slated for removal in Python 3.16
    asyncio.set_event_loop_policy(runner.HassEventLoopPolicy(False))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
--------------------------- snapshot report summary ----------------------------
1 snapshot failed. 12 snapshots passed. 5 snapshots unused.

Re-run pytest with --snapshot-update to delete unused snapshots.
=========================== short test summary info ============================
FAILED tests/components/hypontech/test_sensor.py::test_sensors - AssertionError: assert [+ received] == [- snapshot]
=================== 1 failed, 16 passed, 2 warnings in 1.34s ===================</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>tests.home-assistant-component-tests.hypontech</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.hyponcloud</li>
    <li>python313Packages.hyponcloud.dist</li>
    <li>python314Packages.hyponcloud</li>
    <li>python314Packages.hyponcloud.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>tests.home-assistant-component-tests.hypontech</summary>
<pre>2026-03-22 23:05:26.416 INFO     MainThread homeassistant.helpers.entity_registry:entity_registry.py:1260 Registered new sensor.hypontech entity: sensor.rooftop_power
2026-03-22 23:05:26.418 INFO     MainThread homeassistant.helpers.entity_registry:entity_registry.py:1260 Registered new sensor.hypontech entity: sensor.rooftop_lifetime_energy
2026-03-22 23:05:26.418 INFO     MainThread homeassistant.helpers.entity_registry:entity_registry.py:1260 Registered new sensor.hypontech entity: sensor.rooftop_today_energy
=============================== warnings summary ===============================
homeassistant/runner.py:176
  /build/source/homeassistant/runner.py:176: DeprecationWarning: &#x27;asyncio.DefaultEventLoopPolicy&#x27; is deprecated and slated for removal in Python 3.16
    class HassEventLoopPolicy(asyncio.DefaultEventLoopPolicy):  # type: ignore[name-defined,misc]

tests/conftest.py:155
  /build/source/tests/conftest.py:155: DeprecationWarning: &#x27;asyncio.set_event_loop_policy&#x27; is deprecated and slated for removal in Python 3.16
    asyncio.set_event_loop_policy(runner.HassEventLoopPolicy(False))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
--------------------------- snapshot report summary ----------------------------
1 snapshot failed. 12 snapshots passed. 5 snapshots unused.

Re-run pytest with --snapshot-update to delete unused snapshots.
=========================== short test summary info ============================
FAILED tests/components/hypontech/test_sensor.py::test_sensors - AssertionError: assert [+ received] == [- snapshot]
=================== 1 failed, 16 passed, 2 warnings in 1.11s ===================</pre>
</details>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.hyponcloud</li>
    <li>python313Packages.hyponcloud.dist</li>
    <li>python314Packages.hyponcloud</li>
    <li>python314Packages.hyponcloud.dist</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.hyponcloud</li>
    <li>python313Packages.hyponcloud.dist</li>
    <li>python314Packages.hyponcloud</li>
    <li>python314Packages.hyponcloud.dist</li>
  </ul>
</details>

--- a/pkgs/development/python-modules/hyponcloud/default.nix
+++ b/pkgs/development/python-modules/hyponcloud/default.nix
@@ -1,0 +1,49 @@
+{
+  lib,
+  aiohttp,
+  buildPythonPackage,
+  fetchFromGitHub,
+  mashumaro,
+  pytest-asyncio,
+  pytestCheckHook,
+  setuptools,
+  setuptools-scm,
+}:
+
+buildPythonPackage (finalAttrs: {
+  pname = "hyponcloud";
+  version = "0.9.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "jcisio";
+    repo = "hyponcloud";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-ZFolkhyXauUS+6rbCjSp+5Oxfp3Y7oh8fjXHwDi+zKA=";
+  };
+
+  build-system = [
+    setuptools
+    setuptools-scm
+  ];
+
+  dependencies = [
+    aiohttp
+    mashumaro
+  ];
+
+  nativeCheckInputs = [
+    pytest-asyncio
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [ "hyponcloud" ];
+
+  meta = {
+    description = "Python library for Hypontech Cloud API for solar inverter monitoring";
+    homepage = "https://github.com/jcisio/hyponcloud";
+    changelog = "https://github.com/jcisio/hyponcloud/releases/tag/v${finalAttrs.version}";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.jamiemagee ];
+  };
+})

--- a/pkgs/servers/home-assistant/component-packages.nix
+++ b/pkgs/servers/home-assistant/component-packages.nix
@@ -2665,7 +2665,8 @@
       ];
     "hypontech" =
       ps: with ps; [
-      ]; # missing inputs: hyponcloud
+        hyponcloud
+      ];
     "ialarm" =
       ps: with ps; [
         pyialarm
@@ -7635,6 +7636,7 @@
     "hvv_departures"
     "hydrawise"
     "hyperion"
+    "hypontech"
     "ialarm"
     "iaqualink"
     "ibeacon"

--- a/pkgs/servers/home-assistant/tests.nix
+++ b/pkgs/servers/home-assistant/tests.nix
@@ -167,6 +167,10 @@ let
   };
 
   extraDisabledTestPaths = {
+    hypontech = [
+      # outdated snapshot
+      "tests/components/hypontech/test_sensor.py::test_sensors"
+    ];
     jellyfin = [
       # AssertionError: assert 'audio/x-flac' == 'audio/flac'
       "tests/components/jellyfin/test_media_source.py::test_resolve"

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -7286,6 +7286,8 @@ self: super: with self; {
 
   hyperscan = callPackage ../development/python-modules/hyperscan { };
 
+  hyponcloud = callPackage ../development/python-modules/hyponcloud { };
+
   hypothesis = callPackage ../development/python-modules/hypothesis { };
 
   hypothesis-auto = callPackage ../development/python-modules/hypothesis-auto { };

